test/TestApp.AspNetCore/Program.cs

```diff
@@ -0,0 +1,63 @@
+// <copyright file="Program.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using TestApp.AspNetCore;
+
+var builder = WebApplication.CreateBuilder(args);
+
+// Add services to the container.
+
+builder.Services.AddControllers();
+
+// Learn more about configuring Swagger/OpenAPI at https://aka.ms/aspnetcore/swashbuckle
+builder.Services.AddEndpointsApiExplorer();
+
+builder.Services.AddSwaggerGen();
+
+builder.Services.AddMvc();
+
+builder.Services.AddSingleton<HttpClient>();
+
+builder.Services.AddSingleton(
+    new CallbackMiddleware.CallbackMiddlewareImpl());
+
+builder.Services.AddSingleton(
+    new ActivityMiddleware.ActivityMiddlewareImpl());
+
+var app = builder.Build();
+
+// Configure the HTTP request pipeline.
+if (app.Environment.IsDevelopment())
+{
+    app.UseSwagger();
+    app.UseSwaggerUI();
+}
+
+app.UseHttpsRedirection();
+
+app.UseAuthorization();
+
+app.MapControllers();
+
+app.UseMiddleware<CallbackMiddleware>();
+
+app.UseMiddleware<ActivityMiddleware>();
+
+app.Run();
+
+public partial class Program
+{
```

Comment:
any reason why we need a partial class here?

Reply:
This was taken from suggestion here - https://docs.microsoft.com/en-us/aspnet/core/test/integration-tests?view=aspnetcore-6.0#:~:text=Make%20the%20Program%20class%20public%20using%20a%20partial%20class%20declaration

Reply:
@cijothomas - I removed this partial class and added the explicit public Program class. Let me know if this is better.